src/Microsoft.Azure.WebJobs.Host/Executors/JobHostContextFactory.cs

```diff
@@ -188,8 +188,7 @@ public JobHostContextFactory(IStorageAccountProvider storageAccountProvider, ICo
                         queueConfiguration, exceptionHandler, trace, functions,
                         functionInstanceLogger, functionExecutor);
 
-                    Guid hostInstanceId = Guid.NewGuid();
```

Comment:
E.g. previously one way I would test singleton support was to spin up multiple instances of the same host on a single machine, so I could watch one get the lease, and the other fail to get it. What would happen now? The second host instance would steal the lease from the first? What happens when the first instance loses its lease?